packages/grafana-ui/src/components/DataLinks/DataLinkInput.tsx

```diff
@@ -44,128 +45,130 @@ const getStyles = stylesFactory((theme: GrafanaTheme) => ({
+
+        case 'ArrowDown':
+        case 'ArrowUp':
+          event.preventDefault();
+          const direction = event.key === 'ArrowDown' ? 1 : -1;
+          return setSuggestionsIndex(index => modulo(index + direction, stateRef.current.suggestions.length));
+        default:
+          return next();
+      }
+    }, []);
+
+    useEffect(() => {
+      // Update the state of the link in the parent. This is basically done on blur but we need to do it after
+      // our state have been updated. The duplicity of state is done for perf reasons and also because local
+      // state also contains things like selection and formating.
+      if (prevLinkUrl && prevLinkUrl.selection.isFocused && !linkUrl.selection.isFocused) {
+        stateRef.current.onChange(Plain.serialize(linkUrl));
+      }
+    }, [linkUrl, prevLinkUrl]);
+
+    const onUrlChange = React.useCallback(({ value }: { value: Value }) => {
+      setLinkUrl(value);
+    }, []);
+
+    const onVariableSelect = (item: VariableSuggestion, editor = editorRef.current!) => {
+      const includeDollarSign = Plain.serialize(editor.value).slice(-1) !== '$';
+      if (item.origin !== VariableOrigin.Template || item.value === DataLinkBuiltInVars.includeVars) {
+        editor.insertText(`${includeDollarSign ? '$' : ''}\{${item.value}}`);
+      } else {
+        editor.insertText(`var-${item.value}=$\{${item.value}}`);
+      }
 
-    setLinkUrl(editor.value);
-    setShowingSuggestions(false);
-
-    setSuggestionsIndex(0);
-    onChange(Plain.serialize(editor.value));
-  };
-
-  return (
-    <div
-      className={cx(
-        'gf-form-input',
-        css`
-          position: relative;
-          height: auto;
-        `
-      )}
-    >
-      <div className="slate-query-field">
-        {showingSuggestions && (
-          <Portal>
-            <ReactPopper
-              referenceElement={selectionRef}
-              placement="top-end"
-              modifiers={{
-                preventOverflow: { enabled: true, boundariesElement: 'window' },
-                arrow: { enabled: false },
-                offset: { offset: 250 }, // width of the suggestions menu
-              }}
-            >
-              {({ ref, style, placement }) => {
-                return (
-                  <div ref={ref} style={style} data-placement={placement}>
-                    <DataLinkSuggestions
-                      suggestions={stateRef.current.suggestions}
-                      onSuggestionSelect={onVariableSelect}
-                      onClose={() => setShowingSuggestions(false)}
-                      activeIndex={suggestionsIndex}
-                    />
-                  </div>
-                );
-              }}
-            </ReactPopper>
-          </Portal>
+      setLinkUrl(editor.value);
+      setShowingSuggestions(false);
+
+      setSuggestionsIndex(0);
+      stateRef.current.onChange(Plain.serialize(editor.value));
```

Comment:
@davkal This is the fix for the disappearing title and regex field. If this does not get merged this should be a separate fix at least cause this can affect other parts using this component.